### PR TITLE
Use `console.log` for warnings

### PR DESCRIPTION
**Pull request by thasner**
`console.warn` is just an alias for `console.error` in nodejs

### CODE REVIEW

File: packages/core/src/PackageContext.ts [hunk 1]
```diff
@@ -127,7 +127,7 @@ export class PackageContext implements Context {
       return;
     }
     // tslint:disable-next-line:no-console
-    console.warn(this.getMessage(str, depth));
```
Comment (thasner) [inline packages/core/src/PackageContext.ts:130]:
```
/**
 * The {@link console.warn()} function is an alias for {@link console.error()}.
 */
warn(message?: any, ...optionalParams: any[]): void;
```

Comment (thasner) [reply packages/core/src/PackageContext.ts:130]:
```
/**
 * The `console.debug()` function is an alias for {@link console.log()}.
 */
debug(message?: any, ...optionalParams: any[]): void;
```

Comment (styu) [reply packages/core/src/PackageContext.ts:130]: can we add a comment describing the above so it doesn't accidentally return

File: packages/core/src/PackageContext.ts [hunk 2]
```diff
@@ -127,7 +127,7 @@ export class PackageContext implements Context {
...
+    console.log(this.getMessage(str, depth));
```
Comment (thasner) [inline packages/core/src/PackageContext.ts:130]:
```suggestion
	// In nodejs, `console.warn` is an alias for `console.error`,
	// which is undesirable for unfixable or unimportant warnings. 
    console.log(this.getMessage(str, depth));
```

File: packages/core/src/PackageContext.ts [hunk 3]
```diff
@@ -127,7 +127,9 @@ export class PackageContext implements Context {
       return;
     }
     // tslint:disable-next-line:no-console
-    console.warn(this.getMessage(str, depth));
+	// In nodejs, `console.warn` is an alias for `console.error`,
+	// which is undesirable for unfixable or unimportant warnings.
```
Comment (thasner) [inline packages/core/src/PackageContext.ts:131]:
```suggestion
    // In nodejs, `console.warn` is an alias for `console.error`,
    // which is undesirable for unfixable or unimportant warnings. 
```

File: packages/core/src/PackageContext.ts [hunk 4]
```diff
@@ -127,7 +127,9 @@ export class PackageContext implements Context {
       return;
     }
     // tslint:disable-next-line:no-console
```
Comment (thasner) [inline packages/core/src/PackageContext.ts:129]:
```suggestion
```

File: packages/core/src/PackageContext.ts [hunk 5]
```diff
@@ -127,7 +127,9 @@ export class PackageContext implements Context {
...
-    console.warn(this.getMessage(str, depth));
+    // In nodejs, `console.warn` is an alias for `console.error`,
+    // which is undesirable for unfixable or unimportant warnings.
```
Comment (thasner) [inline packages/core/src/PackageContext.ts:131]:
```suggestion
    // which is undesirable for unfixable or unimportant warnings.
    // tslint:disable-next-line:no-console
```

**Pull request merged**